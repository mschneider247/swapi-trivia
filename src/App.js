import React, { Component } from 'react';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      user: {},
      movies: [],
      currentMovie: {},
      currentCharacters: []
    }
  }

  render() {
    return(
      <h1>Hello</h1>
    )
  }

}

export default App;
