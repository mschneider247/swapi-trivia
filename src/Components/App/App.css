*,
*:after,
*:before {
  box-sizing: border-box;
}

.App {
  background-image: url('../../images/swapi-stars.svg');
  background-size: cover;
  background-repeat: no-repeat;
  color: #ffffff;
  height: 100vh;
  width: 100%;
  text-align: center;
}

h1 {
  font-size: 14rem;
  margin: 0;
}

.yellow-text {
  color: #EEDB00;
}
