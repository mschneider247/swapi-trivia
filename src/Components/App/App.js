import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
import Profile from '../Profile/Profile';
import Menu from '../Menu/Menu';
import  { Route, Link } from 'react-router-dom';
import { getMovies } from '../../apiCalls/apiCalls';

class App extends Component {
  constructor() {
    super();
    this.state = {
      user: null,
      movies: [],
      currentMovie: {},
      currentCharacters: [],
      favoriteCharacters: []
    }
  }

componentDidMount = () => {
  getMovies('https://swapi.co/api/films/')
    .then(films => this.setState({movies: films}))
    .catch(error => console.error('error'))
}

  updateUserState = (userObj) => {
    this.setState({
      user: userObj
    })
  }

  removeUserState = () => {
    this.setState({
      user: null
    })
  }

  render() {
    return(
      <section className='App'>
        <Menu removeUserState={this.removeUserState}/>
        <Route exact path='/' render={() => <Form updateUserState={this.updateUserState}/>} />
        <Route exact path='/movies' render={() => <Profile name={this.state.user.name} quote={this.state.user.quote} status={this.state.user.status}/>} />
        <Route exact path='/movies' render={() => <h2>MOVIES</h2>} />
        <Route exact path='/movies' render={() => <MoviesContainer movies={this.state.movies} user={this.state.user}/>} />
      </section>
    )
  }

}

export default App;
