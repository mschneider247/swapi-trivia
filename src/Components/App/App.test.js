import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot with correct data', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should update user state with data from form', () => {
    const wrapper = shallow(<App />);

    const mockUserObj = {
      name: 'Mike',
      quote: 'May the force be with you',
      status: 'Jedi'
    }

    wrapper.instance().updateUserState(mockUserObj);

    expect(wrapper.state('user')).toEqual(mockUserObj);
  })

})
