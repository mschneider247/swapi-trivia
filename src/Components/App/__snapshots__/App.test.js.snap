// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with correct data 1`] = `
<section
  className="App"
>
  <Route
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/movies"
    render={[Function]}
  />
  <Route
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/movies/:episode_id"
    render={[Function]}
  />
</section>
`;
